### PR TITLE
Removes some useless dependencies

### PR DESCRIPTION
**Pull request by sileht**
Don't require osc_lib
Drop oslo.utils and futurist
Drop oslo.serialization

### CODE REVIEW

File: gnocchiclient/json.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+# under the License.
+import datetime
+import uuid
+
+import numpy
+import six
+import ujson
+
+
+def to_primitive(obj):
```
Comment (jd) [inline gnocchiclient/json.py]: Do we really need our own JSON encoder here? I fail to see if/when/where we send weird things.

Especially around numpy. That sounds bad. We don't depend on numpy here.

File: gnocchiclient/utils.py [hunk 2]
```diff
@@ -223,3 +223,44 @@ def get_client(obj):
         # TODO(sileht): Remove this when OSC is able
         # to install the gnocchi client binary itself
         return obj.app.client
+
+
+TRUE_STRINGS = ('1', 't', 'true', 'on', 'y', 'yes')
+FALSE_STRINGS = ('0', 'f', 'false', 'off', 'n', 'no')
+
+
+def bool_from_string(subject, strict=False, default=False):
```
Comment (jd) [inline gnocchiclient/utils.py]: we used distutils.util.strtobool in gnocchi that's good enough.

File: gnocchiclient/json.py (hunk 1 above)
Comment (sileht) [inline gnocchiclient/json.py]: my bad I have copied/pasted the gnocchi code, nothing is needed here the raw ujson is enough

**Comment by sileht** [comment]
Pull-request updated, HEAD is now ecbc2a81ce0fd4181166cd4335777123d77963c8

File: gnocchiclient/benchmark.py [hunk 3]
```diff
@@ -36,6 +46,7 @@ def _pickle_method(m):
     else:
         return getattr, (m.im_self, m.im_func.func_name)
 
+
```
Comment (jd) [inline gnocchiclient/benchmark.py]: O_O

Comment (sileht) [inline gnocchiclient/benchmark.py]: Most important part of the change, maybe I comment it ?

Comment (sileht) [inline gnocchiclient/benchmark.py]: I was kidding but my flake8 really want this to pass :(

Comment (jd) [inline gnocchiclient/benchmark.py]: Haha

File: gnocchiclient/benchmark.py [hunk 4]
```diff
@@ -60,26 +71,31 @@ def _positive_non_zero_int(argument_value):
...
+    def elapsed(self):
+        return max(0.0, self.started_at - now())
+
+
 def measure_job(fn, *args, **kwargs):
-    # NOTE(sileht): This is not part of BenchmarkPool
     # because we cannot pickle BenchmarkPool class
-    sw = timeutils.StopWatch().start()
+    sw = StopWatch().start()
```
Comment (chungg) [inline gnocchiclient/benchmark.py]: there's no start method

File: gnocchiclient/benchmark.py [hunk 5]
```diff
@@ -12,21 +12,31 @@
...
 import time
 import types
+import warnings
 
 from cliff import show
-import futurist
-from oslo_utils import timeutils
 import six.moves
 
+from gnocchi import utils
```
Comment (chungg) [inline gnocchiclient/benchmark.py]: this seems like a bit much to be pulling in entire gnocchi lib just for client no?

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 1b0d2cb3668f960ddd08d28e7f52ee752c5c4216

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 7426c3d64193e0d8831b9df475bc00f4e90d3f7b

**Comment by chungg** [comment]
can we not depend on entire gnocchi lib?

**Comment by jd** [comment]
@chungg I don't understand what you mean?

**Comment by chungg** [comment]
@jd we're importing gnocchi just to use utils... so if we want to use gnocchiclient we have to have gnocchi on the machine as well... seems like overkill

File: gnocchiclient/benchmark.py (hunk 5 above)
Comment (jd) [inline gnocchiclient/benchmark.py]: Oh I missed your comment. Yeah I think I already noticed that and it's a left over from @sileht

**Comment by sileht** [comment]
Pull-request updated, HEAD is now d92eff8bd9ebd6e7a2691bfbeec592c9d70310ca

**Comment by jd** [comment]
I don't see any change?

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 8f3d4e99802cba147096f7977e9e033092273767

File: gnocchiclient/benchmark.py (hunk 5 above)
Comment (sileht) [inline gnocchiclient/benchmark.py]: Fixed

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 4e43e1f20fafb80c1d18a521adba746c8be4a63b

File: requirements.txt [hunk 6]
```diff
@@ -5,8 +5,8 @@
 pbr>=1.4
 cliff>1.16.0 # Apache-2.0
 osc-lib>=0.3.0 # Apache-2.0
-oslo.utils>=2.0.0 # Apache-2.0
 ujson
 keystoneauth1>=2.0.0
 six
 futurist
+monotonic
```
Comment (jd) [inline requirements.txt]: you miss iso8601

**Comment by sileht** [comment]
Pull-request updated, HEAD is now bc94596c14c83d16a6fe3c7dbeea066256b42bbc

**Pull request merged**